core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -639,18 +642,22 @@ private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOExcept
     // Precompute this because mPos will be updated several times in this function.
     final boolean isInCurrentBlock = pos / mBlockSize == mPos / mBlockSize;
 
-    if (isInCurrentBlock && isReadFromLocalWorker()) {
-      // no need to partial cache the current block, and the seek is within the block
-      // so directly seeks to position.
-      mPos = pos;
-      // updateStreams is necessary when pos = mFileLength.
+    if (isInCurrentBlock) {
+      // the read is within the current block update stream to refresh the streams
       updateStreams();
```

Comment:
What's the invariant for when `updateStreams` should be called?

Reply:
Calling `updateStreams` will fetch the block information and sync the blockInStream for the current position. Since it will invoke RPCs, we want to call this method only on the block that we are about to read.

Reply:
Is there an invariant we can follow, for example whenever we change `mPos`?

Reply:
There are multiple cases:
 - when `mPos` changes and you need to read the data from `mPos`, this will create new instream and cache stream if `mPos` moves to a new block 
 - when the file is to close
 - when a block read is done, to close or cancel the stream